src/umpire/resource/MemoryResourceTypes.hpp

```diff
@@ -106,22 +107,27 @@ inline MemoryResourceType string_to_resource(const std::string& resource)
 
 inline int resource_to_device_id(const std::string& resource)
 {
+  int device_id{0};
+
+#if defined(UMPIRE_ENABLE_CUDA)
+  if (resource.find("::") != std::string::npos) {
+    device_id = std::stoi(resource.substr(resource.find("::") + 2));
+  } else {
+    // get the device bound to the current process
+    cudaGetDevice(&device_id);
+  }
+#elif defined(UMPIRE_ENABLE_HIP)
```

Comment:
CUDA and HIP are not the only device types, so I would just leave this as an `else` rather than `elif`

Reply:
Done.